cpp/src/io/orc/stripe_data.cu

```diff
@@ -782,6 +780,7 @@ static __device__ uint32_t Integer_RLEv2(orc_bytestream_s* bs,
     pos  = shuffle(pos);
     n    = shuffle(n);
     w    = shuffle(w);
+    __syncwarp();
```

Comment:
Not sure this one is needed here, as our `shuffle` is an alias for `__shfl_sync`, which, to my understanding, would converge threads participating in the shuffle (in our case: there is no mask, so all threads participate).
If, despite, `__syncwarp` should be required, we should leave a note that clarifies why we need `__syncwarp` here.

Reply:
I'll add a comment. Really want to go towards error-free memcheck/racecheck reports.

Reply:
This one resolves the following `racecheck` warnings, presumably because the tool does not recognize `shuffle_sync` as a sync point.
```
Warning: Race reported between Write access at 0x19520 in /cudf/cpp/src/io/orc/stripe_data.cu:735:unsigned int Integer_RLEv2
and Read access at 0x19be0 in /cudf/cpp/src/io/orc/stripe_data.cu:807:unsigned int Integer_RLEv2[488 hazards]

Warning: Race reported between Write access at 0x19050 in /cudf/cpp/src/io/orc/stripe_data.cu:773:unsigned int Integer_RLEv2
and Read access at 0x196d0 in /cudf/cpp/src/io/orc/stripe_data.cu:816:unsigned int Integer_RLEv2 [16 hazards]
```